feat: added guide for epp standalone with inferencepool dependency (#2260)
kubernetes-sigs/gateway-api-inference-extension: site-src/guides/epp-standalone.md

## How
A proxy is deployed as a sidecar to the EPP. The proxy and EPP continue to communicate via ext-proc protocol over localhost.
For the endpoint discovery, you have two options:

> * With Inference APIs Support: The InferencePool Custom Resource (CR) provides structured management within the Kubernetes ecosystem and supports the entire suite of Inference APIs, including the use of InferenceObjectives for defining workload priority.

Author: done

> * Without Inference APIs Support: Configure model server pod selectors directly as a flag to the EPP. This is the simplest method for standalone jobs without needing to install the inference extension apis, but it will not support the features that depend on inference APIs.

Author: done

> * Without Inference APIs Support: The EPP is configured using command line flags. This is the simplest method for standalone jobs which doesn't require installing the inference extension apis, which means no support for the features expressed using the inference APIs (such as InferenceObjectives).

Reviewer: ```suggestion
* **Without Inference APIs Support:** The EPP is configured using command line flags. This is the simplest method for standalone jobs which doesn't require installing the inference extension apis, which means no support for the features expressed using the inference APIs (such as InferenceObjectives).
```

Author: done

#### Deploy Endpoint Picker Extension with Envoy sidecar

##### Option 1: Direct Flags

> Deploy an Endpoint Picker Extension named `vllm-llama3-8b-instruct` that selects from endpoints with label `app=vllm-llama3-8b-instruct` and listening on port 8000. The Helm install command automatically installs the endpoint-picker specific resources.

Author: done

#### Deploy Endpoint Picker Extension with Envoy sidecar

<!-- TABS:START -->

<!-- TAB:With Inference APIs Support -->

> ##### Option 1: With Inference APIs Support

Reviewer: not showing up correctly yet, I am expecting them to show side-by-side
<img width="889" height="1085" alt="image" src="https://github.com/user-attachments/assets/9534afa8-6894-4bbe-8792-684bb935b540" />

     oci://us-central1-docker.pkg.dev/k8s-staging-images/gateway-api-inference-extension/charts/epp-standalone
   ```

> ##### Option 2: InferencePool Dependency

Reviewer: We should make this option 1 and explain that with this mode all inference extension apis are supported.

> Install an InferencePool named `vllm-llama3-8b-instruct` that selects from endpoints with label app: vllm-llama3-8b-instruct and 

Reviewer: ```suggestion
Deploy an InferencePool named `vllm-llama3-8b-instruct` that selects from endpoints with label app: vllm-llama3-8b-instruct and 
```

> listening on port 8000. The Helm install command automatically installs the endpoint-picker, InferencePool along with provider specific resources.

Reviewer: ```suggestion
listening on port 8000. The Helm install command automatically deploys an InferencePool instance, the epp along with provider specific resources.
```

Author: done
